tests/test_api.py

```diff
@@ -32,6 +32,9 @@ def test_top_level_api():
 def test_awkward_api():
     assert dir(pylhe.awkward) == ["register_awkward", "to_awkward"]
 
+    with pytest.warns(DeprecationWarning):
+        pylhe.register_awkward()
```

Comment:
This should be its own test. c.f. https://github.com/scikit-hep/pyhf/blob/d6208cbabb8ab727f201e6e06673f938d77b14b2/tests/test_regression.py#L183-L196 as an example, but doesn't need to be as detailed as that test.